src/widgets/hero-course/ui/hero-course.test.tsx

```diff
@@ -0,0 +1,87 @@
+import { screen } from '@testing-library/react';
+import { HeroCourse } from './hero-course';
+import { MOCKED_IMAGE_PATH } from '@/shared/__tests__/constants';
+import { renderWithRouter } from '@/shared/__tests__/utils';
+import { dayJS } from '@/shared/helpers/dayJS';
+
+vi.mock('@/app/hooks/use-course-by-title');
+vi.mock('react-router-dom', async (importOriginal) => {
+  const actual = await importOriginal<typeof import('react-router-dom')>();
+
+  return {
+    ...actual,
+    useLoaderData: vi.fn(() => [mockedCourse, mockedCourseAvailable, mockedCourseUpcoming]),
+  };
+});
+
+const reactCourseTitle = 'React';
+const angularCourseTitle = 'Angular';
+
+const mockedCourse = {
```

Comment:
`const mockedCourse: Course = {`
and the same for other object

Reply:
Actually, it's a part of `Course` type. Should I add to mock all fields or create type for testing with `omit` / `pick`?
![image](https://github.com/user-attachments/assets/eb5bdc69-af07-4949-aba2-7c1dcc8bfb90)

Reply:
I'd prefer to have all the properties just in case we use some of them in our logic, e.g. upcoming-courses tests use the following properties
![image](https://github.com/user-attachments/assets/e3c21803-9a3a-48d8-ad17-1643d5ee540f)
but the typeguard requires the use of some additional props, and because of this the behavior in tests and in a real application is different
![image](https://github.com/user-attachments/assets/89fffa25-49ae-440f-b734-e3361168fff0)

Reply:
Done